hadoop-hdds/common/src/main/java/org/apache/hadoop/utils/db/RDBTable.java

```diff
@@ -183,4 +183,14 @@ public String getName() throws IOException {
   public void close() throws Exception {
     // Nothing do for a Column Family.
   }
+
+  @Override
+  public long getEstimatedKeyCount() throws IOException {
+    try {
+      return db.getLongProperty(handle, "rocksdb.estimate-num-keys");
+    } catch (RocksDBException e) {
+      throw new IOException(
+          "Failed to get estimated key count of table.");
```

Comment:
Does this exception handling look good to you?

Reply:
If we throw an exception here, Ratis will get a failure via the future returned from `notifyInstallSnapshotFromLeader`. I am not sure what the impact of that is. Will Ratis terminate the process? cc @hanishakoneru 

I'd say for now this is okay. I filed https://issues.apache.org/jira/browse/HDDS-1862 to investigate further.

Reply:
Though I'm not sure how the exception would affect Ratis, I found there are a bunch of methods in this class that are converting RocksDBException into IOException as well, like `RDBTable#get`, `RDBTable#delete`, `RDBTable#getName`.